pkg/deployments/manager.go

```diff
@@ -103,6 +113,13 @@ func (r *Manager) addDeployment(ctx context.Context, d appsv1.Deployment) error
 	return nil
 }
 
+func (r *Manager) HasModel(deploymentName string) bool {
+	r.deploymentsMtx.RLock()
+	defer r.deploymentsMtx.RUnlock()
+	_, exists := r.deployments[deploymentName]
+	return exists
+}
+
```

Comment:
Please remove the deployment in `removeDeployment()`

Reply:
done, great catch!